### PR TITLE
Feature/file modularity

### PR DESCRIPTION
**Pull request by alon-david**

### CODE REVIEW

**Comment by idanasulinmemphis** [comment]
Hi @alon-david, this is awesome, we will go over it and will update in the meantime please resolve the conflicts and change back the client.py file to memphis.py, we prefer it that way

**Comment by alon-david** [comment]
fixed, please do not merge anything else before this branch is merged or declined

**Review by idanasulinmemphis — APPROVED** [review]

**Pull request merged**
